presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -342,6 +359,13 @@ public QueryInfo createQuery(SessionSupplier sessionSupplier, String query)
         QueryExecution queryExecution;
         Statement statement;
         try {
+            if (!acceptQueries.get()) {
+                if (nanosSince(initialNanos).compareTo(initializationTimeout) < 0 && internalNodeManager.getNodes(ACTIVE).size() < initialMinWorkersRequired) {
+                    throw new PrestoException(SERVER_STARTING_UP, "Insufficient active worker nodes");
```

Comment:
"Cluster is still initializing, there are insufficient active worker nodes (%number) to run query"